code/modules/clothing/clothing.dm

```diff
@@ -480,6 +481,20 @@
 		"Drask" = 'icons/mob/clothing/species/drask/shoes.dmi'
 		)
 
+/obj/item/clothing/shoes/equipped(mob/user, slot)
+	. = ..()
+	if(!no_slip || slot != SLOT_HUD_SHOES)
+		return
+	ADD_TRAIT(user, TRAIT_NOSLIP, UID())
+
+/obj/item/clothing/shoes/dropped(mob/user)
+	..()
+	if(!no_slip)
+		return
+	var/mob/living/carbon/human/H = user
+	if(H.get_item_by_slot(SLOT_HUD_SHOES) == src)
+		REMOVE_TRAIT(H, TRAIT_NOSLIP, UID())
```

Comment:
Is there a possibility that they will have already been removed from this slot by the time this code is called?

Reply:
I do not think so? I mostly copied the way judo belt/boxing gloves remove a trait, and dropping it seemed to work properly when I was testing it.